scripts/Jenkinsfile

```diff
@@ -0,0 +1,172 @@
+				}
+			}
+		}
+		
+		if (startStage <= 2) { 
+			stage('Stage 2: Initialize GIT submodules') { 
+				if (!fileExists("hcc_fork")) {
+					sh (script:"git clone ssh://git@github.com/${ userID}/hcc.git hcc_fork")
+					sh (script:"cd hcc_fork; git checkout clang_tot_upgrade; git submodule update --init")
+				}
+			}
+		}
+		
+		if (startStage <= 3) { 
+			stage ('Stage 3: Add all remotes and forks for submodules') {
+				git_remote_add("hcc_fork","hcc_fork", "ssh://git@github.com/${ userID}/hcc.git")
+				git_remote_add("hcc_fork/lld","lld_fork", "ssh://git@github.com/${ userID}/lld.git")
+				git_remote_add("hcc_fork/compiler","llvm_fork", "ssh://git@github.com/${ userID}/llvm.git")
+				git_remote_add("hcc_fork/compiler-rt","compiler-rt_fork", "ssh://git@github.com/${ userID}/compiler-rt.git")
+				git_remote_add("hcc_fork/compiler-rt","compiler-rt", "ssh://git@github.com/llvm-mirror/compiler-rt.git")
+				git_remote_add("hcc_fork/clang-tools-extra","clang-tools-extra_fork", "ssh://git@github.com/${ userID}/clang-tools-extra.git")
+				git_remote_add("hcc_fork/clang-tools-extra","clang-tools-extra", "ssh://git@github.com/llvm-mirror/clang-tools-extra.git")
+				git_remote_add("hcc_fork/clang","clang_fork", "ssh://git@github.com/${ userID}/hcc-clang-upgrade.git")
+				git_remote_add("hcc_fork/clang","clang", "ssh://git@github.com/RadeonOpenCompute/clang")
+			}
+		}
+		
+		if (startStage <= 4) { 
+			stage ('Stage 4:Checkout latest master ROCM-Device-Libs') {
+				sh (script: "cd hcc_fork/rocdl; git fetch --all; git checkout origin/master")
+			}
+		}
+		
+		if (startStage <= 5) { 		
+			stage ('Stage 5: Merge amd-common  commits') {
+			    
+			    if (!fileExists("merge-branch-name") ) { 
+			    	writeFile file: "merge-branch-name", text: "${ merge_name}"
+				} 
+				merge_name = readFile "merge-branch-name"
+				
+				dir("hcc_fork") {
+					git_config()
+					git_merge("lld", merge_name, "origin/amd-hcc", "origin/amd-common")
+					git_merge("compiler",merge_name,"origin/amd-hcc", "origin/amd-common")
+					git_merge("compiler-rt", merge_name, "origin/amd-hcc", "--no-ff compiler-rt/master")
+					git_merge("clang-tools-extra", merge_name, "origin/amd-hcc", "clang-tools-extra/master")
+					git_merge("clang", merge_name, "origin/clang_tot_upgrade", "--no-ff clang/amd-common")
+				}
+			}		
+		}
+		
+		if (startStage <= 6) { 
+			stage ('Stage 6: Build merged HCC') {
+				sh (script: "mkdir -p build; cd build; cmake -DCMAKE_BUILD_TYPE=Release ../hcc; make -j10")
+			}
+		}
+		
+		if (startStage <= 7) { 
+			stage ('Stage 7: Full sanity tests on merged HCC') {
+				sh (script: "cd build; make test")
+			}
+		}
+		
+		if (startStage <= 8) { 		
+			stage ('Stage 8: Quick sanity tests on merged HCC') {
+				sh (script: "/opt/rocm/bin/hcc `/opt/rocm/bin/clamp-config --build --cxxflags --ldflags` -lm hcc/tests/Unit/AmpMath/amp_math_cos.cpp; ./a.out ")
```

Comment:
@david-salinas  I'm not very familiar with Jenkins syntax, for quick sanity tests, would the return value for `./a.out` be checked?

Reply:
Yes. The return value from the shell script would be returned to Jenkins at this line, by default.  I can add a hook at the end of the sh () command here to handle specific return values.  But for now they are just passed up to Jenkins.  Any non-zero RC will throw an exception.  And I have a generic catch at the bottom of all of these steps.  But we can extend this if need be.